### PR TITLE
COMPASS-4109: disable connect button when syntax error is present

### PR DESCRIPTION
**Pull request by alenakhineika**
## Description

Disable the connect button in case of syntax errors and allow to connect only when URI is valid.

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [x] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/components/form/form-actions.spec.js [hunk 1]
```diff
@@ -161,6 +166,14 @@ describe('FormActions [Component]', () => {
...
         expect(component.find(classname)).to.be.present();
       });
+
+      it('disables the connect button', () => {
+        const connectButton = component
+          .find('button[name="connect"]')
+          .simulate('click');
+
+        expect(connectButton.hasClass('disabled')).to.equal(true);
+      });
```
Comment (alenakhineika) [inline src/components/form/form-actions.spec.js]: The test that checks if the button is disabled when a syntax error is present.

File: src/components/form/form-actions.jsx [hunk 2]
```diff
@@ -127,12 +130,15 @@ class FormActions extends React.Component {
       <button
         type="submit"
         name="connect"
-        className="btn btn-sm btn-primary"
-        onClick={this.onConnectClicked.bind(this)}>
+        className={`btn btn-sm btn-primary ${
+          this.hasSyntaxError() ? 'disabled' : ''
+        }`}
+        onClick={this.onConnectClicked.bind(this)}
+      >
```
Comment (alenakhineika) [inline src/components/form/form-actions.jsx]: If there is a syntax error disable the button.

File: src/components/form/form-actions.spec.js (hunk 1 above)
Comment (lrlna) [reply src/components/form/form-actions.spec.js]: ssdw

Comment (lrlna) [reply src/components/form/form-actions.spec.js]: lol sorry that was Nori ^. I guess she approves too

**Pull request merged**